Speed up rebuilding classic hex from string

vihangm opened the pull request:
Using string stream is extremely slow.
Instead this just iterates over the string and decodes each hex character as we go.

Benchmark results from running google benchmark on the side:
```
------------------------------------------------------------
Benchmark                  Time             CPU   Iterations
------------------------------------------------------------
BM_SoleStream            623 ns          623 ns      1117378
BM_SoleIter             91.2 ns         91.2 ns      7631954
```

Also added a benchmark to this repo, here are the before and after:
```
String Stream:
Benchmarking rebuild... 1464148 uuids/sec
Iteration:
Benchmarking rebuild... 17086712 uuids/sec
```

r-lyeh commented:
Cool thx! :D Any volunteers to test this? I will merge within the next days if no issues are reported in this thread

vihangm commented:
Realized that the rebuild benchmark I added was timing both the call to `str` and the `rebuild`. Fixed it to stringify outside the benchmark and the improvement now seems to line up with what google benchmark reports.

vihangm commented:
Friendly ping. Let me know if there's any tests (besides the included ones) that I can help run and get this merged. :)

r-lyeh commented:
No probs. If someone complains we can also fix it later
